lib/Onyx.js

```diff
@@ -801,22 +803,58 @@ function disconnect(connectionID, keyToRemoveFromEvictionBlocklist) {
     delete callbackToStateMapping[connectionID];
 }
 
+// [key, value, canUpdateSubscriber]
+let pendingUpdates = [];
+let scheduledPromise;
+let pendingCollectionUpdates = [];
+
+/**
+ * We are batching together onyx updates. This helps with use cases where we schedule onyx updates after each other.
+ * This happens for example in the Onyx.update function, where we process API responses that might contain a lot of
+ * update operations. Instead of calling the subscribers for each update operation, we batch them together which will
+ * cause react to schedule the updates at once instead of after each other. This is mainly a performance optimization.
+ * @returns {Promise}
+ */
+function scheduleOnyxFlushIfNeeded() {
+    if (scheduledPromise) {
+        return scheduledPromise;
+    }
+
+    scheduledPromise = new Promise((resolve) => {
+        setTimeout(() => {
+            scheduledPromise = null;
+            const updatesCopy = pendingUpdates;
+            pendingUpdates = [];
+            const updatesCollectionCopy = pendingCollectionUpdates;
+            pendingCollectionUpdates = [];
+            unstable_batchedUpdates(() => {
+                for (let i = 0; i < updatesCopy.length; ++i) {
+                    keyChanged(updatesCopy[i][0], updatesCopy[i][1], updatesCopy[i][2]);
```

Comment:
I couldn't find anything that confirms that the `pendingUpdate|updatesCopy` is has a 3rd child. Presumably, this is guaranteed as that's how our Onyx updates are written?

Reply:
![Screenshot 2023-09-05 at 09 39 03](https://github.com/Expensify/react-native-onyx/assets/12784455/c3375c41-ce1c-4aa0-9f37-3b4860ffefd7)
The keyChanged method has 3 arguments so we only delay the execution.